UnitTests/GitCommandsTests/ExternalLinks/ConfiguredLinkDefinitionsProviderTests.cs

```diff
@@ -48,12 +48,11 @@ public void Setup()
         [TearDown]
         public void TearDown()
         {
-            _userRoaming.SettingsCache.Dispose();
-            _repoDistributed.SettingsCache.Dispose();
-            _repoLocal.SettingsCache.Dispose();
+            _userRoaming.SettingsCache?.Dispose();
+            _repoDistributed.SettingsCache?.Dispose();
+            _repoLocal.SettingsCache?.Dispose();
```

Comment:
Why would these be `null`?

Reply:
`Dispose` should dispose as much as possible. In C++, destructors even must not throw exceptions.
The constructor could throw half-way. Isn't `Dispose` called anyway then?

Reply:
I don't believe so. If ctor throws, there is no instance to dispose even though a memory may have been allocated.